envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/InternalRedirectTest.kt

```diff
@@ -0,0 +1,64 @@
+package pl.allegro.tech.servicemesh.envoycontrol
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlRunnerTestApp
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
+import pl.allegro.tech.servicemesh.envoycontrol.config.redirect.RedirectServiceContainer
+import java.net.UnknownHostException
+
+internal class InternalRedirectTest : EnvoyControlTestConfiguration() {
```

Comment:
To be honest I think this test class should be open, so users can remodel the expectations in their implementation of this feature.